### PR TITLE
Feature/quotation trade open tab

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x] UpdateFormModal
- [x] TradeTab
- [x] TradeTabMobile
- [x] PositionOpenModal
- [x] PositionClosedModal

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/trade_tab/trade_tab.tsx [hunk 1]
```diff
@@ -439,56 +499,28 @@ const TradeTab = () => {
...
   const getToggledShortSlSetting = (bool: boolean) => {
     setShortSlToggle(bool);
 
-    setExpectedShortLossValue(
-      (shortSlValue - Number(shortQuotationRef.current?.price)) * targetInputValueRef.current
-    );
+    calculateShortLoss();
   };
 
   const toApplyCreateOrder = (): {
```
Comment (arealclimber) [inline src/components/trade_tab/trade_tab.tsx:526]: - 送出給用戶簽的畫面，要先從 context 拿報價單再顯示

File: src/components/trade_tab/trade_tab.tsx [hunk 2]
```diff
@@ -223,164 +247,207 @@ const TradeTab = () => {
...
-          typeOfPosition: TypeOfPosition.BUY,
-          unitAsset: unitAsset,
-          price: buyPrice,
-          deadline: getTimestamp() + QUOTATION_RENEWAL_INTERVAL_SECONDS,
-          signature: '0x',
-        };
-
-        setLongQuotation(buyQuotation);
-      }
+  const setQuotation = () => {
```
Comment (arealclimber) [inline src/components/trade_tab/trade_tab.tsx:250]: - 不用拚報價單，只需要算價格

File: src/components/trade_tab/trade_tab.tsx [hunk 3]
```diff
@@ -176,19 +169,48 @@ const TradeTab = () => {
...
+      marketCtx.selectedTicker?.price,
+      'long',
+      longQuotationRef.current,
+      'short',
+      shortQuotationRef.current
+    );
+  }, [marketCtx.selectedTicker?.price]);
+
   // Info: Fetch quotation in period (20230327 - Shirley)
   useEffect(() => {
```
Comment (arealclimber) [inline src/components/trade_tab/trade_tab.tsx:215]: - 不用擔心到期

File: src/components/open_sub_tab/open_sub_tab.tsx [hunk 4]
```diff
@@ -43,187 +47,20 @@ const OpenSubTab = () => {
...
       const positionLineGraph = marketCtx.listTickerPositions(cfd.targetAsset, {
         begin: cfd.createTimestamp,
       });
-      const quotation =
-        cfd.typeOfPosition === TypeOfPosition.BUY
-          ? longQuotationRef.current
-          : shortQuotationRef.current;
 
-      const displayCFD: IDisplayCFDOrder = toDisplayCFDOrder(cfd, positionLineGraph, quotation);
+      const sd = cfd.typeOfPosition === TypeOfPosition.BUY ? 1 + spread : 1 - spread;
```
Comment (arealclimber) [inline src/components/open_sub_tab/open_sub_tab.tsx:60]: // TODO: 沒有折線圖，就拿市價來算；沒有市價就拿顯示 `--`

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**